### PR TITLE
Migration of loading and favoriteitems to ES6 modules

### PR DESCRIPTION
**Pull request by Camc314**
**Issues**
part of #868

### CODE REVIEW

File: src/components/favoriteitems.js [hunk 1]
```diff
@@ -223,7 +230,8 @@ define(['loading', 'libraryBrowser', 'cardBuilder', 'dom', 'apphost', 'imageLoad
         });
     }
 
-    return {
-        render: loadSections
-    };
-});
+export default {
+    render: loadSection
```
Comment (dkanada) [inline src/components/favoriteitems.js:234]:
```suggestion
    render: loadSections
```

File: src/components/loading/loading.js [hunk 2]
```diff
@@ -1,78 +1,74 @@
...
+let layer2;
+let layer3;
+let layer4;
+let circleLefts;
+let circleRights;
 
-    return {
-        show: function () {
-            var elem = loadingElem;
+function show() {
```
Comment (dmitrylyzo) [inline src/components/loading/loading.js:11]: _not tested_
```suggestion
export function show() {
```

File: src/components/loading/loading.js [hunk 3]
```diff
@@ -1,78 +1,74 @@
...
+    }
+}
 
-            for (i = 0, length = circleRights.length; i < length; i++) {
-                circleRights[i].classList.add('mdl-spinner__circleRight-active');
-            }
-        },
-        hide: function () {
-            var elem = loadingElem;
+function hide() {
```
Comment (dmitrylyzo) [inline src/components/loading/loading.js:50]: _not tested_
```suggestion
export function hide() {
```

File: src/components/appRouter.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ define(['loading', 'globalize', 'events', 'viewManager', 'skinManager', 'backdro
 
     function beginConnectionWizard() {
         backdrop.clearBackdrop();
-        loading.show();
+        loading.default.show();
```
Comment (dmitrylyzo) [inline src/components/appRouter.js:25]: `loading = loading.default || loading;`?

File: src/components/loading/loading.js (hunk 3 above)
Comment (Camc314) [reply src/components/loading/loading.js:50]: If you want, I can add it, but it works with/without it

File: src/components/appRouter.js (hunk 4 above)
Comment (Camc314) [reply src/components/appRouter.js:25]: Updated

**Pull request merged**